tests/ui/manual_split_once.rs

```diff
@@ -41,13 +41,107 @@ fn main() {
     let _ = s.rsplitn(2, '=').nth(1);
 }
 
+fn indirect() -> Option<()> {
+    let mut iter = "a.b.c".splitn(2, '.');
+    let l = iter.next().unwrap();
+    let r = iter.next().unwrap();
+
+    let mut iter = "a.b.c".splitn(2, '.');
+    let l = iter.next()?;
+    let r = iter.next()?;
+
+    let mut iter = "a.b.c".rsplitn(2, '.');
+    let r = iter.next().unwrap();
+    let l = iter.next().unwrap();
+
+    let mut iter = "a.b.c".rsplitn(2, '.');
+    let r = iter.next()?;
+    let l = iter.next()?;
+
+    // could lint, currently doesn't
+
+    let mut iter = "a.b.c".splitn(2, '.');
+    let other = 1;
+    let l = iter.next()?;
+    let r = iter.next()?;
+
+    let mut iter = "a.b.c".splitn(2, '.');
+    let mut mut_binding = iter.next()?;
+    let r = iter.next()?;
+
+    let mut iter = "a.b.c".splitn(2, '.');
+    let tuple = (iter.next()?, iter.next()?);
```

Comment:
Can we also add the limitations to the lint's documentation?

Something like [this](https://github.com/rust-lang/rust-clippy/blob/cbdf17c884116f17b1efe8c6dd3c41eb45d1342c/clippy_lints/src/enum_variants.rs#L21-L25)

Reply:
We can add that. I usually go by feel if a limitation has to be added to the description or not. Having it in the documentation is never wrong (If it's up-to-date).

Could you also add the limitation, that only `next()` calls directly following the `split()` call get linted?

Reply:
Added!